migrations/educandu-2022-09-07-02-streamline-nulls-in-plugins.js

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable camelcase, no-console, no-await-in-loop, complexity */
+
+import { EFFECT_TYPE } from '../src/plugins/image/constants.js';
+import { IMAGE_SOURCE_TYPE, MEDIA_ASPECT_RATIO } from '../src/domain/constants.js';
+import { SOUND_SOURCE_TYPE, TEST_MODE } from '../src/plugins/ear-training/constants.js';
+
+export default class Educandu_2022_09_07_02_streamline_nulls_in_plugins {
+  constructor(db) {
+    this.db = db;
+  }
+
+  tryUpdateSection(section) {
+    let sectionWasUpdated = false;
+
+    if (!section.content) {
+      return sectionWasUpdated;
+    }
+
+    if (section.type === 'markdown' && 'renderMedia' in section.content) {
+      // Delete superfluous property
+      delete section.content.renderMedia;
+      sectionWasUpdated = true;
+    }
+
+    if (section.type === 'annotation' && 'renderMedia' in section.content) {
+      // Delete superfluous property
+      delete section.content.renderMedia;
+      sectionWasUpdated = true;
+    }
+
+    if (section.type === 'table' && 'renderMedia' in section.content) {
+      // Delete superfluous property
+      delete section.content.renderMedia;
+      sectionWasUpdated = true;
+    }
+
+    if (section.type === 'interactive-media' && 'range' in section.content) {
+      // Delete wrongly named property
+      delete section.content.range;
+      sectionWasUpdated = true;
+    }
+
+    if (section.type === 'image') {
+      if (section.content.copyrightNotice === null) {
+        // Convert null -> empty string
+        section.content.copyrightNotice = '';
+        sectionWasUpdated = true;
+      }
+      if (section.content.effect?.type === EFFECT_TYPE.clip && 'copyrightNotice' in section.content.effect) {
+        // Delete superfluous property
+        delete section.content.effect.copyrightNotice;
+        sectionWasUpdated = true;
```

Comment:
With these effects, maybe it would have been better to have:
- either dedicated objects
- and the `effect.type` outside of the `effect` object
- both of the above
Let's see if it is worth changing this at some point..

Reply:
I thought the same (towards `effect` never being `null` but always an object (even if one containing only `type: 'none'`). Let's create a ticket for that, it will also make the editor less complicated!

Reply:
I created a ticket.

Reply:
👍